controllers/managed_scaledown_test.go

```diff
@@ -0,0 +1,324 @@
+				Expect(k8sClient.Create(ctx, loggingConfigMap)).Should(Succeed())
+				brokerCrd.Spec.DeploymentPlan.ExtraMounts.ConfigMaps = []string{loggingConfigMapName}
+
+				booleanTrue := true
+				brokerCrd.Spec.DeploymentPlan.Clustered = &booleanTrue
+				brokerCrd.Spec.DeploymentPlan.Size = common.Int32ToPtr(3)
+				brokerCrd.Spec.DeploymentPlan.PersistenceEnabled = true
+				brokerCrd.Spec.BrokerProperties = []string{
+					"HAPolicyConfiguration=PRIMARY_ONLY",
+					// use the cluster discovery group for scaledown
+					"HAPolicyConfiguration.scaleDownConfiguration.discoveryGroup=my-discovery-group",
+					// this makes sig term or forceFailover do scaledown and is the trigger for using condition based scaledown
+					// 	const ScaleDownConfigTrigger = "HAPolicyConfiguration.scaleDownConfiguration.enabled=true"
+					ScaleDownConfigTrigger,
+				}
+				Expect(k8sClient.Create(ctx, brokerCrd)).Should(Succeed())
+
+				createdBrokerCrd := &brokerv1beta1.ActiveMQArtemis{}
+				By("verifying ready")
+				Eventually(func(g Gomega) {
+
+					getPersistedVersionedCrd(brokerCrd.ObjectMeta.Name, defaultNamespace, createdBrokerCrd)
+					if verbose {
+						fmt.Printf("\nSTATUS:%v\n", createdBrokerCrd.Status)
+					}
+
+					By("Check ready status")
+					g.Expect(len(createdBrokerCrd.Status.PodStatus.Ready)).Should(BeEquivalentTo(3))
+					g.Expect(meta.IsStatusConditionTrue(createdBrokerCrd.Status.Conditions, brokerv1beta1.DeployedConditionType)).Should(BeTrue())
+					g.Expect(meta.IsStatusConditionTrue(createdBrokerCrd.Status.Conditions, brokerv1beta1.ReadyConditionType)).Should(BeTrue())
+
+				}, existingClusterTimeout, existingClusterInterval).Should(Succeed())
+
+				pod0BeforeScaleDown := &corev1.Pod{}
+				podWithOrdinal0 := namer.CrToSS(brokerName) + "-0"
+				Expect(k8sClient.Get(ctx, types.NamespacedName{Name: podWithOrdinal0, Namespace: defaultNamespace}, pod0BeforeScaleDown)).Should(Succeed())
+
+				podWithOrdinal1 := namer.CrToSS(brokerName) + "-1"
+
+				By("Sending a message to 1")
+				Eventually(func(g Gomega) {
+
+					sendCmd := []string{"amq-broker/bin/artemis", "producer", "--user", "Jay", "--password", "activemq", "--url", "tcp://" + podWithOrdinal1 + ":61616", "--message-count", "1", "--destination", "queue://DLQ", "--verbose"}
+					content, err := RunCommandInPod(podWithOrdinal1, brokerName+"-container", sendCmd)
+					g.Expect(err).To(BeNil())
+					g.Expect(*content).Should(ContainSubstring("Produced: 1 messages"))
+
+				}, timeout, interval).Should(Succeed())
+
+				By("Scaling down to ss-0")
+				Eventually(func(g Gomega) {
+
+					getPersistedVersionedCrd(brokerCrd.ObjectMeta.Name, defaultNamespace, createdBrokerCrd)
+					createdBrokerCrd.Spec.DeploymentPlan.Size = common.Int32ToPtr(1)
+					g.Expect(k8sClient.Update(ctx, createdBrokerCrd)).Should(Succeed())
+					g.Expect(meta.IsStatusConditionTrue(createdBrokerCrd.Status.Conditions, brokerv1beta1.ScaleDownPendingConditionType)).Should(BeTrue())
```

Comment:
The update could be executed multiple times because ScaleDownPendingConditionType is not true or  the ScaleDownPendingConditionType could have already been removed.